test/new-e2e/tests/containers/base_test.go

```diff
@@ -575,3 +575,116 @@ func (suite *baseSuite[Env]) testEvent(args *testEventArgs) {
 		}, 2*time.Minute, 10*time.Second, "Failed finding `%s` with proper tags and message", prettyEventQuery)
 	})
 }
+
+type testHostTags struct {
+	ExpectedTags *[]string
+	OptionalTags *[]string
+}
+
+func sendEvent[Env any](suite *baseSuite[Env], alertType, text string, args *testHostTags) {
+	formattedArgs, err := yaml.Marshal(args)
+	suite.Require().NoError(err)
+
+	_, err = suite.DatadogClient().PostEvent(&datadog.Event{
+		Title:     pointer.Ptr("test Host-Tags " + suite.T().Name()),
+		AlertType: &alertType,
+		Tags: append([]string{
+			"app:agent-new-e2e-tests-containers",
+			"cluster_name:" + suite.clusterName,
+			"test:" + suite.T().Name(),
+		}, *args.ExpectedTags...),
+		Text: pointer.Ptr(fmt.Sprintf(
+			`%%%%%%
+### Result
+
+`+"```"+`
+%s
+`+"```"+`
+
+### Expected Tags
+
+`+"```"+`
+%s
+`+"```"+`
+%%%%%%
+`, text, formattedArgs)),
+	})
+
+	if err != nil {
+		suite.T().Logf("Failed to post event: %s", err)
+	}
+}
+
+func getLatestHostInfos(hostInfos []*aggregator.HostTags) *aggregator.HostTags {
+	if len(hostInfos) <= 0 {
+		return nil
+	}
+
+	var latestHostInfo = hostInfos[0]
+
+	for _, host := range hostInfos {
+		if latestHostInfo.GetCollectedTime().Before(host.GetCollectedTime()) {
+			// current `host` is more recent, keep it
+			latestHostInfo = host
+		}
+	}
+
+	return latestHostInfo
```

Comment:
I would be surprised if the fake intake wasn’t returning the payloads exactly in the same order as the one in which it received them.

So, that the most recent one should be retrievable with a simple:


```suggestion
	return hostInfos[len(hostInfos)-1]
```

Reply:
make sense, I'll try, I hope you're right. otherwise we'll see it quickly the actual test will be super flaky !